source/windows/ipc_util.c

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <windows.h>
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/ipc_util.h>
+#include <aws/common/logging.h>
+#include <inttypes.h>
+
+struct aws_ipc_util_instance_lock {
+    struct aws_allocator *allocator;
+    HANDLE mutex;
+};
+
+struct aws_ipc_util_instance_lock *aws_ipc_util_instance_lock_try_acquire(
+    struct aws_allocator *allocator,
+    struct aws_byte_cursor instance_nonce) {
+
+    /* validate we don't have a directory slash. */
+    struct aws_byte_cursor to_find = aws_byte_cursor_from_c_str("\\");
+    struct aws_byte_cursor found;
+    AWS_ZERO_STRUCT(found);
+    if (aws_byte_cursor_find_exact(&instance_nonce, &to_find, &found) != AWS_OP_ERR &&
+        aws_last_error() != AWS_ERROR_STRING_MATCH_NOT_FOUND) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        return NULL;
+    }
+
+    struct aws_byte_cursor path_prefix = aws_byte_cursor_from_c_str("Global/");
```

Comment:
trivial: Not sure why we'd bother with a prefix, but make it totally generic.

If we worry the nonce won't be unique enough, then make this more unique, like "AwsCrtGlobal/"

But if we think the nonce will already take care to be unique, then a prefix isn't necessary

Reply:
Ah, Global is a thing to inform windows how to scope the lock. I think it probably should be local though, so I'll test that.  I think we want the session namespace. The /tmp path already does that due to the way the /tmp directory is mounted.

from msdn: 
"The name can have a "Global" or "Local" prefix to explicitly create the object in the global or session namespace. The remainder of the name can contain any character except the backslash character (\). "